rust/src/io/reader.rs

```diff
@@ -1131,4 +1131,56 @@ mod tests {
             &expected_large_list
         );
     }
+
+    #[tokio::test]
+    async fn test_list_array_with_offsets() {
+        let arrow_schema = ArrowSchema::new(vec![
+            ArrowField::new(
+                "l",
+                DataType::List(Box::new(ArrowField::new("item", DataType::Int32, true))),
+                false,
+            ),
+            ArrowField::new(
+                "ll",
+                DataType::LargeList(Box::new(ArrowField::new("item", DataType::Int32, true))),
+                false,
+            ),
+        ]);
+
+        let store = ObjectStore::memory();
+        let path = Path::from("/lists");
+
+        let list_array = ListArray::from_iter_primitive::<Int32Type, _, _>(vec![
+            Some(vec![Some(1), Some(2)]),
+            Some(vec![Some(3), Some(4)]),
+            Some((0..1_000_000).map(|n| Some(n)).collect::<Vec<_>>()),
+        ])
+        .slice(1, 1);
+        println!("test: list_array: {:?}", list_array);
+        let large_list_array = LargeListArray::from_iter_primitive::<Int32Type, _, _>(vec![
+            Some(vec![Some(10), Some(11)]),
+            Some(vec![Some(12), Some(13)]),
+            Some((0..1_000_000).map(|n| Some(n)).collect::<Vec<_>>()),
```

Comment:
Do we have a test to actually write multiple arrays, wanna just make sure the cummulation of offsets is correct